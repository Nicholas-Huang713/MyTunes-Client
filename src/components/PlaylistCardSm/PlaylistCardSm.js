import React, {useState} from 'react'
import PlayCircleFilledIcon from '@material-ui/icons/PlayCircleFilled';
import './PlaylistCardSm.scss';

export default function PlaylistCardSm(props) {
    const [hovered, setHovered] = useState(false);

    return (
        <div 
            data-testid="card-sm"
            className="playlist-card-sm" 
            onMouseEnter={() => setHovered(true)}
            onMouseLeave={() => setHovered(false)}
            // onDoubleClick= {() => }
        >
            <div className="playlist-card-sm-img mr-3 ml-1">
                <img alt="album cover art" src={props.song.preview} />
            </div>
            <div className="playlist-card-sm-title">
                {props.title}
            </div>
            {hovered && 
                <PlayCircleFilledIcon data-testid='play-button' fontSize="large" className="playlist-card-sm-playicon"/>
            }
        </div>
    )
}
