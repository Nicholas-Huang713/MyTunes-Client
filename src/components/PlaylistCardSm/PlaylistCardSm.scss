@import '../../styles/config'; 


.playlist-card-sm {
    width: 31%;
    display: flex;
    border:1px solid lightblue;
    margin:0.5rem;
    border-radius: 0.4rem;
    background: $color-primary;
    color: $color-secondary;
    padding: 10px;
    // div {
    //     padding: 10px;
    // }
    &:hover {
        background-color: gray;
    }

    .playlist-card-sm-img {
        // flex-grow: 1;
    }
    .playlist-card-sm-title {
        // flex-grow: 9;
        display: flex;
        align-items: center;
        justify-content: flex-start;
        align-content: space-between;
    }

}

.playlist-card-sm-playicon {
    // width: 200px;
    position: absolute;
    // top: 1px;
    // left: 5;
    // right: 0.5;

}