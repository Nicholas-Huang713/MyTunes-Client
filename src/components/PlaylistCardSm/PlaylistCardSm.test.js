import { render, cleanup, fireEvent} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PlaylistCardSm from './PlaylistCardSm';

afterEach(cleanup);

const props = { title: "My Playlist 1", song: {preview: "/img.jpg"}}

describe('<PlaylistCardSm />', () => {

    it('should match snapshot', () => {
        const { asFragment } = render(<PlaylistCardSm {...props}/>);
        expect(asFragment()).toMatchSnapshot();
      });

    it('should render with props', () => {
        
        const {getByAltText, getByText, queryByTestId} = render(<PlaylistCardSm {...props}/>);
        expect(getByAltText('album cover art').getAttribute('src')).toBe('/img.jpg');
        expect(getByText('My Playlist 1')).toBeInTheDocument()
        expect(queryByTestId('play-button')).not.toBeInTheDocument()
    });

    it('should render play button when hovered', async () => {
        const {queryByTestId} = render(<PlaylistCardSm {...props}/>);
        // let trigger = 
        expect(queryByTestId('play-button')).not.toBeInTheDocument()
        const trigger = queryByTestId('card-sm');
        userEvent.hover(trigger);
        expect(queryByTestId('play-button')).toBeInTheDocument()
        userEvent.unhover(trigger);
        expect(queryByTestId('play-button')).not.toBeInTheDocument()
    });
    

    
});
