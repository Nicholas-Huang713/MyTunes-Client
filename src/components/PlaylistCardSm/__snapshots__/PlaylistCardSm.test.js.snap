// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaylistCardSm /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="playlist-card-sm"
    data-testid="card-sm"
  >
    <div
      class="playlist-card-sm-img mr-3 ml-1"
    >
      <img
        alt="album cover art"
        src="/img.jpg"
      />
    </div>
    <div
      class="playlist-card-sm-title"
    >
      My Playlist 1
    </div>
  </div>
</DocumentFragment>
`;
